src/server/ua_server.c

```diff
@@ -241,12 +241,43 @@ void UA_Server_delete(UA_Server *server) {
                     &UA_TYPES[UA_TYPES_ENDPOINTDESCRIPTION]);
 
 #ifdef UA_ENABLE_DISCOVERY
-    registeredServer_list_entry *current, *temp;
-    LIST_FOREACH_SAFE(current, &server->registeredServers, pointers, temp) {
-        LIST_REMOVE(current, pointers);
-        UA_RegisteredServer_deleteMembers(&current->registeredServer);
-        UA_free(current);
+    {
```

Comment:
I used the additional blocks to keep `*current` and `*temp` local. IMHO it does not make much sense to move the functionality in those blocks into a small function, because it's less than 20 lines and it is only cleaning up